File: server/PQL-Domain-Question/src/main/java/com/passql/question/dto/SubmitResult.java
package com.passql.question.dto;

import java.util.UUID;

// EXECUTABLE 문제: selectedResult/correctResult/selectedSql/correctSql 포함
// CONCEPT_ONLY 문제: 위 4개 필드 null
// submissionUuid: 저장된 제출 UUID — 프론트 신고 기능에서 사용
public record SubmitResult(
    boolean isCorrect,
    String correctKey,
    String rationale,
    ExecuteResult selectedResult,
    ExecuteResult correctResult,
    String correctSql,
    String selectedSql,
    UUID submissionUuid
) {}

File: app/lib/data/models/question/submit_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'execute_result.dart';

part 'submit_result.freezed.dart';
part 'submit_result.g.dart';

/// 제출 결과. ResultPage로 navigate state에 담아 전달.
@freezed
class SubmitResult with _$SubmitResult {
  const factory SubmitResult({
    required bool isCorrect,
    String? correctKey,
    String? rationale,
    ExecuteResult? selectedResult,
    ExecuteResult? correctResult,
    String? correctSql,
    String? selectedSql,
  }) = _SubmitResult;

  factory SubmitResult.fromJson(Map<String, dynamic> json) =>
      _$SubmitResultFromJson(json);
}
